### PR TITLE
machine: update to use new-style sd-bus macros

### PR DESCRIPTION
**Pull request by amarjargal**
Replace old SD_BUS_METHOD_WITH_NAMES and SD_BUS_SIGNAL_WITH_NAMES macros to
the new SD_BUS_METHOD_WITH_ARGS and SD_BUS_SIGNAL_WITH_ARGS macros.

Meson test, mkosi test image and running machinectl after build returned
no error. But since I don't have any virtual machines or containers, I'm not
sure how to test the changes thoroughly.

### CODE REVIEW

File: src/machine/machine-dbus.c [hunk 1]
```diff
@@ -1278,85 +1278,61 @@ static const sd_bus_vtable machine_vtable[] = {
...
+                                bus_machine_method_open_pty,
+                                SD_BUS_VTABLE_UNPRIVILEGED),
+        SD_BUS_METHOD_WITH_ARGS("OpenLogin",
+                                SD_BUS_NO_ARGS,
+                                SD_BUS_RESULT("h", pty, "s", pty_path),
+                                bus_machine_method_open_login,
+                                SD_BUS_VTABLE_UNPRIVILEGED),
+        SD_BUS_METHOD_WITH_ARGS("OpenShell",
+                                SD_BUS_ARGS("s", user, "s", path, "as", args, "as", environment),
+                                SD_BUS_RESULT("h", pty, "s",pty_path),
```
Comment (daandemeyer) [inline src/machine/machine-dbus.c]: Missing a space here before `pty_path`

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
Looks good - when pushing the style fix requested above, also please amend the commit title as such: `machine: update to use new-style sd-bus macros`

**Comment by amarjargal** [comment]
> Looks good - when pushing the style fix requested above, also please amend the commit title as such: `machine: update to use new-style sd-bus macros`

Got it!

File: src/machine/machine-dbus.c (hunk 1 above)
Comment (amarjargal) [reply src/machine/machine-dbus.c]: I fixed it in another commit.

**Comment by bluca** [comment]
> > Looks good - when pushing the style fix requested above, also please amend the commit title as such: `machine: update to use new-style sd-bus macros`
> 
> Got it!

Thanks - no need to change it now, but for next time, please amend the existing commit when fixing things up, rather than adding a new commit, and then force push (eg: https://www.educative.io/edpresso/how-to-change-a-git-commit-message-after-a-push )

**Comment by keszybz** [comment]
`update-dbus-docs` yields no changes, so this should be good.

**Pull request merged**